pkg/build/generator/generator.go

```diff
@@ -337,6 +338,15 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 		},
 	}
 
+	if binary != nil {
+		build.Spec.Source.Git = nil
+		build.Spec.Source.Binary = binary
+		build.Spec.Source.Type = buildapi.BuildSourceBinary
+		if build.Spec.Source.Dockerfile != nil && binary.AsFile == "Dockerfile" {
```

Comment:
this seems like a weird special case.  if i have a buildconfig that explicitly defines a Dockerfile, and then i do a binary build in which i explicitly pass a Dockerfile, then, and only then, does the Dockerfile i explicitly pass take precedence?

but if my binary archive includes a Dockerfile, it doesn't take precedence?